### DISCUSSION TITLE
How can you express a multi-key object via an object comprehension?

### DISCUSSION THREAD

**Discussion by bleggett** [open-policy-agent/community#197] [OPA and Rego] [answered]
So I wanna use array and object comprehension to build a list of objects, but each object has multiple keys and for the life of me the only way I can figure out how to get this to work is via using `object.merge` to merge two single-kv objects, like so:

```
obj_list := [st | st = object.union(
	{key:val| key = "entity_identifier"; val = attrs[i].entity_identifier}, 
	{key2:val2| key2 = "entity_attributes"; val2 = [{
			"attribute": "https://example.org/attr/OPA/value/AddedByOPA",
			"displayName": "Added By OPA"
			}]}
); attrs[i]]
```

Am a missing a way to express multi-property objects in a single object comprehension, or is the above the only way to do this?

**Comment by anderseknert** [comment, marked as answer]
Hi @bleggett! Maybe I'm missing something too :) Is there some reason you need to use object comprehensions at all here?

```rego
obj_list := [obj | 
    some attr in attrs
    obj := {
    	"entity_identifier": attr.entity_identifier,
        "entity_attributes": [{
	    "attribute": "https://example.org/attr/OPA/value/AddedByOPA",
            "displayName": "Added By OPA"
        }]
}]
```

**Reply by bleggett** [reply]
@anderseknert AH! Thank you - I had missed the existence of the `in` keyword (and the need to `import future.keywords.in` to get it to work)

Correct, much simpler. Thanks much!
